src/components/UI/SearchInput.js

```diff
@@ -0,0 +1,53 @@
+import styles from "./SearchInput.module.css";
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+
+const DEFAULT_DELAY = 1000;
+
+const SearchInput = ({ onChange, delay }) => {
+  const [value, setValue] = useState("");
+  const [isEmpty, setIsEmpty] = useState(true);
+  const timeoutId = useRef();
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeoutId.current);
+    };
+  }, []);
+
+  const changeWithDebounce = (value) => {
+    setValue(value);
+    setIsEmpty(!value);
+
+    clearTimeout(timeoutId.current);
+
+    timeoutId.current = setTimeout(() => {
+      onChange(value);
+    }, delay ?? DEFAULT_DELAY);
+  };
+
+  const changeHandler = (event) => {
+    changeWithDebounce(event?.target?.value ?? "");
+  };
+  const clearButtonClickHandler = () => {
+    changeWithDebounce("");
+  };
+
+  return (
+    <div className={styles.wrapper}>
+      <input value={value} onChange={changeHandler} className={styles.search} />
+      <button
+        className={styles.iconButton}
+        onClick={!isEmpty ? clearButtonClickHandler : undefined}
+      >
+        <i
+          class={`fa-solid ${
+            !isEmpty ? "fa-circle-xmark" : "fa-magnifying-glass"
```

Comment:
isEmpty ?  "fa-magnifying-glass" : "fa-circle-xmark"

Reply:
also you can use array of consts outside the component
```
const BUTTON_ICONS = {
  'false': 'fa-circle-xmark',
  'true': 'fa-magnifying-glass',
}

```

then code will be someting like 

```
class={`fa-solid ${BUTTON_ICONS[isEmpty]`} 
```